kork-pubsub-aws/src/main/java/com/netflix/spinnaker/kork/pubsub/aws/SQSSubscriberProvider.java

```
@@ -0,0 +1,118 @@
+import com.netflix.spinnaker.kork.pubsub.aws.api.AmazonPubsubMessageHandlerFactory;
+import com.netflix.spinnaker.kork.pubsub.aws.config.AmazonPubsubProperties;
+import com.netflix.spinnaker.kork.pubsub.model.PubsubSubscriber;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import javax.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+
+/** * Starts the individual SQS workers (one for each subscription) */
+@Component
+@ConditionalOnExpression("${pubsub.enabled:false} && ${pubsub.amazon.enabled:false}")
+public class SQSSubscriberProvider implements EurekaActivated {
+  private static final Logger log = LoggerFactory.getLogger(SQSSubscriberProvider.class);
+
+  private final AWSCredentialsProvider awsCredentialsProvider;
+  private final AmazonPubsubProperties properties;
+  private final PubsubSubscribers pubsubSubscribers;
+  private final AmazonPubsubMessageHandlerFactory pubsubMessageHandlerFactory;
+  private final Registry registry;
+
+  @Autowired
+  public SQSSubscriberProvider(
+      AWSCredentialsProvider awsCredentialsProvider,
+      AmazonPubsubProperties properties,
+      PubsubSubscribers pubsubSubscribers,
+      AmazonPubsubMessageHandlerFactory pubsubMessageHandlerFactory,
+      Registry registry) {
+    this.awsCredentialsProvider = awsCredentialsProvider;
+    this.properties = properties;
+    this.pubsubSubscribers = pubsubSubscribers;
+    this.pubsubMessageHandlerFactory = pubsubMessageHandlerFactory;
+    this.registry = registry;
+  }
+
+  @PostConstruct
+  public void start() {
+    if (properties == null) {
+      return;
```

I feel like this should throw an exception, or at minimum log an error. If you've got the distance of setting up configuration to the point that `@ConditionalOnExpression("${pubsub.enabled:false} && ${pubsub.amazon.enabled:false}")` evaluates true, I would think the application is misconfigured and should not silently fail to do what it's supposedly enabled to do.